### PR TITLE
Port gz-cmake to use colcon on Windows

### PR DESCRIPTION
**Pull request by j-rivero**
This is the last gz- of #412. ign-cmake was a bit special since it adds a CMake parameters to the compilation thus I needed to implement that parameter passing in the form of `_COLCON_EXTRA_CMAKE_ARGS`.

If we want to keep the history of the CI jobs, we can rename them before we merge this PR. //cc @Blast545. This also affects t https://github.com/osrf/buildfarmer/pull/338.

Test:

 * New ign_cmake-ci-win [Build Status](https://build.osrfoundation.org/job/ign_cmake-ci-win/9/)
 * No use of cmake argument ign-plugin [Build Status](https://build.osrfoundation.org/job/ign_plugin-ci-win/31/)

### CODE REVIEW

**Review by scpeters — APPROVED** [review]

**Review by Blast545 — APPROVED** [review]
LGTM, if you can rename the job before merging I'd highly appreciate it :pray:

**Comment by j-rivero** [comment]
Merged an renamed the -ci- jobs for the supported branches. I left the previous -pr- job in place so the current links in CI to it can work. New one should be in use from now on, that can duplicate the CI reports but I think that is better than losing the links.

**Pull request merged**
